### PR TITLE
camelCase - submitFormAjax.js

### LINKED ISSUE #9151: camelCase submit_form_ajax.js filename
Hi, this is a [first-timers-only issue](https://code.publiclab.org/#r=all). This means we've worked to make it more legible to folks who either **haven't contributed to our codebase before, or even folks who haven't contributed to open source before**. 

If that's you, we're interested in helping you take the first step and can answer questions and help you out as you do. Note that we're especially interested in contributions from people from groups underrepresented in free and open source software!

We know that the process of creating a pull request is the biggest barrier for new contributors. This issue is for you 💝

If you have contributed before, **consider leaving this one for someone new**, and looking through our general [help wanted](https://github.com/publiclab/plots2/labels/help-wanted) issues. Thanks!

### 🤔 What you will need to know.

Nothing. This issue is meant to welcome you to Open Source :) We are happy to walk you through the process.

### ❓ Why are we doing this?

See Issue #9140 for more information!

### 📋 Step by Step

- [ ] 🙋 **Claim this issue**: Comment below. If someone else has claimed it, ask if they've opened a pull request already and if they're stuck -- maybe you can help them solve a problem or move it along!

- [ ] 📝 **Update** these files below:

[See this page](https://code.publiclab.org/#r=all) for some help in taking your first steps!

Below are some "diffs" showing in red (and a `-`) which lines to remove, and in green (and a `+`) which lines to add for each file:

`/app/assets/javascripts/application.js`
```diff
+ //= require submitFormAjax.js
- //= require submit_form_ajax.js
```

`/app/assets/javascripts`:
```diff
+ submitFormAjax.js
- submit_form_ajax.js
```

- [ ] 💾 **Commit** your changes

- [ ] 🔀 **Start a Pull Request**. There are two ways how you can start a pull request:

1. If you are familiar with the terminal or would like to learn it, [here is a great tutorial](https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github) on how to send a pull request using the terminal.

2. You can also [edit files directly in your browser](https://help.github.com/articles/editing-files-in-your-repository/) and open a pull request from there. 

- [ ] 🏁 **Done** Ask in comments for a review :)

### Please keep us updated

💬⏰ - We encourage contributors to be respectful to the community and provide an update **within a week** of claiming a first-timers-only issue. We're happy to keep it assigned to you as long as you need if you update us with a request for more time or help, but if we don't see any activity a week after you claim it we may reassign it to give someone else a chance. Thank you in advance! 

If this happens to you, don't sweat it! Grab another open issue.

### Is someone else already working on this?

🔗- We encourage contributors to link to the original issue in their pull request so all users can easily see if someone's already started on it. 

👥- **If someone seems stuck, offer them some help!** Otherwise, [take a look at some other issues you can help with](https://code.publiclab.org/#r=all). Thanks!

### 🤔❓ Questions?

Leave a comment below!

### PR DESCRIPTION
**Pull request by StreakSharn**
Fixes #9151 (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [ ] code is in uniquely-named feature branch and has no merge conflicts 📁
* [ ] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by noi5e** [comment]
@StreakSharn This looks great! It looks like [codeclimate is showing "2 issues to fix"](https://codeclimate.com/github/publiclab/plots2/pull/9157), which is why this one isn't ready-to-merge.

It turns out that codeclimate is running against some old code in our repository and won't let this pass until it's changed. Our bad, not yours!

Can you do us a favor and change the code in `app/assets/javascripts/submitFormAjax.js` also in this PR?

**Comment by noi5e** [comment]
Instead of `!!responseEl`: `responseEl`

Instead of `callback && callback(event)`: `if (callback) { callback(event); }`

**Comment by StreakSharn** [comment]
Does this work any better?

File: app/assets/javascripts/submitFormAjax.js [hunk 1]
```diff
@@ -8,10 +8,10 @@ function sendFormSubmissionAjax(dataObj, submitTo, responseEl = "", callback) {
     data: dataObj
   })
   .done((event) =>{
-    if (!!responseEl) {
+    if (responseEl) {
       $(responseEl).trigger('ajax:success', event);
     }
-    callback && callback(event);
+    if (callback) { callback(event); };
```
Comment (Sagarpreet) [inline app/assets/javascripts/submitFormAjax.js:14]:
```suggestion
    if (callback) { callback(event); }
    return;
```

@noi5e we can simply `return` or do we have to return output of `callback ` function?

Comment (noi5e) [reply app/assets/javascripts/submitFormAjax.js:14]: @Sagarpreet, no need to return the output of `callback`. This is just calling the function.

I'm pretty sure that we don't need to do an empty `return` here either&mdash;this will work without it.

I think the thing to do here is to remove the unnecessary semicolon, and just override codeclimate's other suggestion.

I know that @jywarren is able to override codeclimate and merge this!

**Comment by jywarren** [comment]
Done! All good and merging. Yes, another reason maybe to think about moving to another review bot is that we've never found a way to share access to overriding CodeClimate!

All done though, thank you all!!!

**Comment by noi5e** [comment]
Nice! @StreakSharn congrats!!!
